kotlin/workflow-core/src/main/java/com/squareup/workflow/Worker.kt

```diff
@@ -178,12 +178,33 @@ interface Worker<out T> {
      *   emitOutput(3)
      * }
      * ```
+     *
+     * Note: If your worker just need to perform side effects and doesn't need to emit anything,
+     * use [createSideEffect] instead (since `Nothing` can't be used as a reified type parameter).
      */
     inline fun <reified T> create(
       key: String = "",
       noinline block: suspend Emitter<T>.() -> Unit
     ): Worker<T> = TypedWorker(T::class, key, block)
 
+    /**
+     * Creates a [Worker] that just performs some side effects and doesn't emit anything.
+     *
+     * The returned [Worker] will equate to any other workers created with this function that have
+     * the same key.
+     *
+     * E.g.:
+     * ```
+     * fun logOnEntered(message: String) = Worker.createSideEffect("logOnEntered") {
+     *   println("Entered state: $message")
+     * }
+     * ```
+     */
+    fun createSideEffect(
```

Comment:
Should add a test.

Reply:
For what? There's no logic here it's just a convenience wrapper.

Reply:
So that we'll know if we manage to break the wrapper, and know that the block is actually called.